test/smoke/src/main.ts

```diff
@@ -299,14 +300,28 @@ describe(`VSCode Smoke Tests (${opts.web ? 'Web' : 'Electron'})`, () => {
 	}
 
 	describe(`VSCode Smoke Tests (${opts.web ? 'Web' : 'Electron'})`, () => {
-		before(async function () {
-			const app = new Application(this.defaultOptions);
-			await app!.start(opts.web ? false : undefined);
-			this.app = app;
-		});
-
-		after(async function () {
-			await this.app.stop();
+		before(function () {
+			const suites = this.test?.parent?.suites || [];
+			for (const suite of suites) {
+				// Add beforeAll hook
+				suite.beforeAll(async function () {
+					const app = new Application(this.defaultOptions);
+					await app!.start(opts.web ? false : undefined);
+					this.app = app;
+				});
+
+				// Move beforeAll to the front of the hook list
+				const suiteBeforeAll = suite['_beforeAll'].pop();
+				suite['_beforeAll'].unshift(suiteBeforeAll);
```

Comment:
Is this stable enough not to break if we update mocha? Are we allowed to dynamically mess with the suites? Shouldn't we simply find a different way of doing this, like passing along `before` and `after` functions to the `setup*` functions? Eg:

```ts
const before = () => {...};
const after = () => {...};
const opts = { before, after };

setupDataLossTests(opts);
setupDataPreferencesTests(opts);
// ...
```